RELEASES.md

```diff
@@ -0,0 +1,84 @@
+# GUAC release process
+
+This document includes information about the GUAC release process. Including
+the release cadence and the process to perform releases.
+
+## Release cadence
+
+The GUAC release cadence aims to happen monthly. 
```

Comment:
move to bi-monthly to start till we start to automate, and then create a request to release if needed. (TODO: create a template to request a release).